packages/enzyme-adapter-react-17/src/ReactSeventeenAdapter.js

```diff
@@ -0,0 +1,893 @@
+  // React stores references to the Provider's defaultValue differently across versions.
+  if ('_defaultValue' in Provider._context) {
+    return Provider._context._defaultValue;
+  }
+  if ('_currentValue' in Provider._context) {
+    return Provider._context._currentValue;
+  }
+  throw new Error('Enzyme Internal Error: can’t figure out how to get Provider’s default value');
+}
+
+function makeFakeElement(type) {
+  return { $$typeof: Element, type };
+}
+
+function isStateful(Component) {
+  return Component.prototype && (
+    Component.prototype.isReactComponent
+    || Array.isArray(Component.__reactAutoBindPairs) // fallback for createClass components
+  );
+}
+
+class ReactSeventeenAdapter extends EnzymeAdapter {
+  constructor() {
+    super();
+    const { lifecycles } = this.options;
+    this.options = {
+      ...this.options,
+      enableComponentDidUpdateOnSetState: true, // TODO: remove, semver-major
+      legacyContextMode: 'parent',
+      lifecycles: {
+        ...lifecycles,
+        componentDidUpdate: {
+          onSetState: true,
+        },
+        getDerivedStateFromProps: {
+          hasShouldComponentUpdateBug: false,
+        },
+        getSnapshotBeforeUpdate: true,
+        setState: {
+          skipsComponentDidUpdateOnNullish: true,
+        },
+        getChildContext: {
+          calledByRenderer: false,
+        },
+        getDerivedStateFromError: true,
+      },
+    };
+  }
+
+  createMountRenderer(options) {
+    assertDomAvailable('mount');
+    if (has(options, 'suspenseFallback')) {
+      throw new TypeError('`suspenseFallback` is not supported by the `mount` renderer');
+    }
+    if (FiberTags === null) {
+      // Requires DOM.
+      FiberTags = detectFiberTags();
+    }
+    const { attachTo, hydrateIn, wrappingComponentProps } = options;
+    const domNode = hydrateIn || attachTo || global.document.createElement('div');
+    let instance = null;
+    const adapter = this;
+    return {
+      render(el, context, callback) {
+        return wrapAct(() => {
+          if (instance === null) {
+            const { type, props, ref } = el;
+            const wrapperProps = {
+              Component: type,
+              props,
+              wrappingComponentProps,
+              context,
+              ...(ref && { refProp: ref }),
+            };
+            const ReactWrapperComponent = createMountWrapper(el, { ...options, adapter });
+            const wrappedEl = React.createElement(ReactWrapperComponent, wrapperProps);
+            instance = hydrateIn
+              ? ReactDOM.hydrate(wrappedEl, domNode)
+              : ReactDOM.render(wrappedEl, domNode);
+            if (typeof callback === 'function') {
+              callback();
+            }
+          } else {
+            instance.setChildProps(el.props, context, callback);
+          }
+        });
+      },
+      unmount() {
+        ReactDOM.unmountComponentAtNode(domNode);
```

Comment:
In a PR to unofficial React 17 adapter, https://github.com/wojtekmaj/enzyme-adapter-react-17/pull/22, the following was suggested by @Iskander508:

```suggestion
        wrapAct(() => {
          ReactDOM.unmountComponentAtNode(domNode);
        });
```